Dangerfile

```diff
@@ -102,6 +115,27 @@ unless changelog_warnings.empty?
   end
 end
 
+
+
+################################################
+# Check Documentation TOC update
+doc_dir_structure_modified = (git.added_files + git.deleted_files + git.renamed_files.map(&:values)).any? { |path| path.start_with?('Documentation/') }
+if doc_dir_structure_modified
+  doc_files = Dir.chdir('Documentation') { Dir['{Articles/,}*{/,.md}'] } - ['README.md']
+  toc_links = File.read('Documentation/README.md').scan(%r{\[.*\]\((.*)\)}).uniq.map { |s| CGI.unescape(s.first) }
+
+  (doc_files - toc_links).each do |doc_not_linked|
+    # Some docs that are not present in links: warn to add them
+    fail("The documentation file #{github.html_link(doc_not_linked)} is not referenced in the #{github.html_link('Documentation/README.md')}. Please add a link to it.")
+  end
+  (toc_links - doc_files).each do |bad_link|
+    fail("The #{github.html_link('Documentation/README.md')} contains a link to \`#{bad_link}\` but that file doesn't exist. Maybe it was renamed or deleted?")
+  end
+end
```

Comment:
Tested on the current state of the PR (which forgot to add the new doc to the TOC) but also by adding a voluntary typo in a link in the `Documentation/README.md` + renaming a `Documentation/Articles/*.md` file, which would lead to this kind of Danger messages:

<img width="914" alt="image" src="https://user-images.githubusercontent.com/216089/87078041-5e01aa00-c224-11ea-84a1-a3a3ed1a3bcc.png">

Note that I could maybe have implemented a more advanced logic to detect git renames (`git.renamed_files`) and try to generate a message like "file X got renamed from OldName to NewName but TOC still references the OldName"… but felt it wasn't worth the trouble of complexifying the script.